amd/src/userlist_updater.js

```diff
@@ -0,0 +1,91 @@
+ */
+define(['jquery', 'core/ajax'], function($, Ajax) {
+    var t = {
+        /**
+         * Initialise the handling.
+         */
+        init: function() {
+            $('select#id_groupid').on('change', t.groupChanged);
+            t.lastGroup = null;
+            t.courseid = $('input:hidden[name=course]').val();
+        },
+
+        /**
+         * Source of data for Ajax element.
+         */
+        groupChanged: function() { 
+            if ($('select#id_groupid').val() === t.lastGroup) {
+                return;
+            }
+            t.lastGroup = $('select#id_groupid').val();
+
+            if (t.lastGroup === '') {
+                t.updateChoices([]);
+            } else {
+                Ajax.call([{
+                    methodname: 'mod_accredible_reload_users',
+                    args: { courseid: t.courseid, groupid: $('select#id_groupid').val()}
+                }])[0].done(t.updateChoices);
+            }
+        },
+
+        /**
+         * Update the contents of the User list with the results of the AJAX call.
+         *
+         * @param {Array} response - array of users.
+         */
+        updateChoices: function(response) {
+            var output = "";
+            var userselements = $('#users-container .form-group').not('.femptylabel');
+            var userscontainer = $('#users-container');
+
+            userselements.remove();
+
+            $(response).each(function(index, option) {
+                if (option.credential_url) {
+                    output += "<div id='fitem_id_certlink" + option.id + "' class='form-group row fitem'>";
+                    output += "<div class='col-md-3 col-form-label d-flex pb-0 pr-md-0'>";
+                    output += "<span class='d-inline-block'>" + option.name + "   " + option.email + "</span></div>";
+                    output += "<div class='col-md-9 form-inline align-items-start felement' data-fieldtype='static'>";
+                    output += "<div class='form-control-static'>Certificate " + option.credential_id;
+                    output += " - <a href='" + option.credential_url + "' target='_blank'>link</a></div></div></div>";
+                } else {
+                    output += "<div class='form-group row fitem checkboxgroup1'>";
+                    output += "<div class='col-md-3'></div><div class='col-md-9 checkbox'>";
+                    output += "<div class='form-check d-flex'>";
+                    output += "<input type='hidden' name='users[" + option.id + "]' value='0'>";
+                    output += "<input type='checkbox' name='users[" + option.id + "]'"; 
+                    output += "class='form-check-input checkboxgroup1' value='1' id='id_users_" + option.id + "'>";
+                    output += "<label for='id_users_" + option.id + "'>";
+                    output += option.name + "   " + option.email;
+                    output += "</label></div></div></div>";
```

Comment:
Question - Did you find any examples that append HTML elements like this to `mod_form` with jQuery? I was a bit worried about adding them without the built-in function `$mform->addElement`.

Reply:
I found this [plugin](https://github.com/moodleou/moodle-filter_embedquestion/blob/main/amd/src/questionid_choice_updater.js) which I use as an example. It doesn't add new inputs as we are doing, but adding new options to a select.

I thought the same as you, that maybe we could generate the new inputs with `addElement` and then append them with the javascript, but I didn't know exactly how to do that.

Reply:
If we make sure how `$mform->addElement` works with other themes, such as using a different theme won't affect expected css styles, I would be fine with the current solution for now. We should definitely consider using the same logic to display the same HTML in a separate ticket.

Reply:
I investigated how `$mform->addElement` works. I haven't completely understood it yet but I believe `$mfrom->addElement` which calls `HTML_QuickForm::addElement()` ([here](https://pear.php.net/manual/en/package.html.html-quickform.html-quickform.addelement.php)) is using mustache templates. For example:
- advcheckbox: https://github.com/moodle/moodle/blob/1a9bee69e60f687bcaf4db0569665f4186c515a4/lib/form/templates/element-advcheckbox.mustache
- static: https://github.com/moodle/moodle/blob/511a87f5fc357f18a4c53911f6e6c7f7b526246e/lib/form/templates/element-static.mustache

As you can see from the history of the files, the output HTML will change when templates are updated so we should use them in `amd/src/userlist_updater.js` as well instead of inserting static HTML.

It looks like mustache templates can be rendered from JavaScript so it would not be very difficult to replace it with the templates.
- https://docs.moodle.org/dev/Templates#Rendering_in_JavaScript
- https://docs.moodle.org/dev/User:Damyon_Wiese/Templates

Reply:
We will fix this point with another ticket NTGR-679. Until then, we won't merge this PR to `develop`.